test/models/commits_test.rb

```diff
@@ -11,6 +11,8 @@ class CommitsTest < ActiveSupport::TestCase
     end
 
     test '.create_from_github handle unknown users' do
+      FakeWeb.register_uri(:get, "https://api.github.com/user/emails", status: %w(200 OK), body: {}.to_json, content_type: 'application/json')
```

Comment:
Again, I think the default config in the test suite should make it so that email filtering isn't enabled, hence that we don't need to fake these calls.

Either that or the calls should be stubbed globally so that we don't have to copy that over in dozens of tests.

Reply:
Right, my apologies for not addressing this correctly earlier.

I wanted to avoid adding more branching logic on whether or not to make the `github_api.emails` request in User, as that begets further complexities and discussions about exposing control of that logic to integrations, so I'll go with a singular register call in the setup for the test here if that's OK.
A more global stub could be done in `test_helper.rb` but I don't think it's necessary / I think it's better for tests to catch any new unaccounted-for interaction with the API in other tests.

In the case of users_test, each test that interacts with the API is setting a specific return value germane to the test logic, so I'm going to leave that as is.

Reply:
> I wanted to avoid adding more branching logic on whether or not to make the `github_api.emails` request in User

It really isn't that much more complexity, and would save useless requests if no org domain is set.

Reply:
(sorry for the bricks of text, I just want to make sure I'm getting things right)

Just to make sure I'm reading things right, we want to make the request behaviour configurable/disable-able because the requests would be useless in the situation where:
1) No org domain is set for the integration.
2) The user(s) have no public email.
3) We specifically don't care about private emails for that project.

That is, in these tests, the (fake) request is made not because of filtering being enabled, but because the fallback logic is to check /emails for a valid email if the public /user did not provide one. Org domain filtering just makes it more likely that the public email is seen as invalid, but in the case of these tests, the email is invalid because it's blank as part of the test.

I can add a flag for projects to configure whether that fallback logic should be happen, but I'll need to figure out where and how far that flag would be exposed (e.g. in shipit.yml, env vars, etc).

I assumed that was too much extra complexity for this change, but maybe that's just because I haven't done that before.

Reply:
Well, I had something much simpler in mind. Basically if you don't restrict emails to a specific domain, it mean you don't really care about private/public emails, hence you can simply take the primary public one directly accessible on the user record.

Reply:
I see, I was unclear before and in the PR description.

The way I interpreted the source issue is that org domain filtering is an additional, optional constraint - and when we don't do that filtering, we still care about trying our best to get _some_ email, whether it's public or private.

Then in the case where we do want a specific org email, we'd rather have no email than a non-org domain email.